yarn-project/pxe/src/kernel_prover/bb_prover/bb_native_proof_creator.ts

```diff
@@ -524,6 +528,28 @@ export class BBNativeProofCreator implements ProofCreator {
     this.log.info(`Successfully verified ${circuitType} proof in ${result.duration} ms`);
   }
 
+  private async verifyProofFromKey(
+    verificationKey: Buffer,
+    proof: Proof,
+    logFunction: (message: string) => void = () => {},
+  ) {
+    // Create random directory to be used for temp files
+    const bbWorkingDirectory = `${this.bbWorkingDirectory}/${randomBytes(8).toString('hex')}`;
+    await fs.mkdir(bbWorkingDirectory, { recursive: true });
+
+    const proofFileName = `${bbWorkingDirectory}/proof`;
+    const verificationKeyPath = `${bbWorkingDirectory}/vk`;
+
+    await fs.writeFile(proofFileName, proof.buffer);
+    await fs.writeFile(verificationKeyPath, verificationKey);
+
+    const result = await verifyProof(this.bbBinaryPath, proofFileName, verificationKeyPath!, logFunction);
```

Comment:
Shall we catch the error and always delete the directory?

Reply:
Yes